In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import nltk
from nltk.corpus import stopwords
from nltk.tokenize import word_tokenize
from nltk.stem import PorterStemmer, WordNetLemmatizer
import re
from sklearn.model_selection import train_test_split

In [2]:
stopwords = stopwords.words('english')

In [3]:
df = pd.read_csv('fake_news_data.csv')
df.head()

title  \
0  HOLLYWEIRD LIB SUSAN SARANDON Compares Muslim ...   
1   Elijah Cummings Called Trump Out To His Face ...   
2   Hillary Clinton Says Half Her Cabinet Will Be...   
3  Russian bombing of U.S.-backed forces being di...   
4  Britain says window to restore Northern Irelan...   

                                                text                 date  \
0  There are two small problems with your analogy...         Dec 30, 2015   
1  Buried in Trump s bonkers interview with New Y...        April 6, 2017   
2  Women make up over 50 percent of this country,...       April 26, 2016   
3  WASHINGTON (Reuters) - U.S. Defense Secretary ...  September 18, 2017    
4  BELFAST (Reuters) - Northern Ireland s politic...   September 4, 2017    

  fake_or_factual  
0       Fake News  
1       Fake News  
2       Fake News  
3    Factual News  
4    Factual News

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 198 entries, 0 to 197
Data columns (total 4 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   title            198 non-null    object
 1   text             198 non-null    object
 2   date             198 non-null    object
 3   fake_or_factual  198 non-null    object
dtypes: object(4)
memory usage: 6.3+ KB


## Discover target value column:

In [5]:
df['fake_or_factual'].value_counts() * 100 / df.shape[0]

Factual News    50.505051
Fake News       49.494949
Name: fake_or_factual, dtype: float64

We have a nearly balanced target value classes.
50.5% Factual News and 49.5% Fake News.

In [6]:
df['title'][1]

' Elijah Cummings Called Trump Out To His Face And Here’s The Pathetic Way Trump Tried To Spin It'

In [7]:
df['text'][1]

'Buried in Trump s bonkers interview with New York Times reporters Maggie Haberman and Glenn Thrush is this outrageous exchange Trump claims happened recently:TRUMP: [U.S. Representative] Elijah Cummings was in my office and he said,  You will go down as one of the great presidents in the history of our country. HABERMAN: Really?TRUMP: And then he went out and I watched him on television yesterday and I said,  Was that the same man? Which  would have been odd considering Cummings, a fiery Democratic congressman, has been one of Trump s most intense critics. He s very public about it. Cummings has been outspoken about Trump s disturbing connections to Russia, the GOP s botched attempts to cover those connections up, and has repeatedly demanded Trump divest himself of his numerous conflicts of interest. He hardly sounds like a person who has looked at Trump s trainwreck first three months in office and concluded that he deserves to be up on Mount Rushmore.One theory was that Trump flatly

In [8]:
def lower_remove_stopwords(text_series):
    return text_series.str.lower().apply(lambda s: ' '.join([w for w in s.split() if w not in stopwords]))

In [9]:
df['text'] = lower_remove_stopwords(df['text'])

In [10]:
df['text'][1]

'buried trump bonkers interview new york times reporters maggie haberman glenn thrush outrageous exchange trump claims happened recently:trump: [u.s. representative] elijah cummings office said, go one great presidents history country. haberman: really?trump: went watched television yesterday said, man? would odd considering cummings, fiery democratic congressman, one trump intense critics. public it. cummings outspoken trump disturbing connections russia, gop botched attempts cover connections up, repeatedly demanded trump divest numerous conflicts interest. hardly sounds like person looked trump trainwreck first three months office concluded deserves mount rushmore.one theory trump flatly lied. always possibility trump pathological liar. another, suggested writer jamelle bouie, trump mixed two black men, conflating cummings guy. another strong possibility trump shown complete lack interest lives black americans.fortunately, cummings came provide real answer: trump took criticism cumm